src/eureka/S1_detector_processing/ramp_fitting.py

```diff
@@ -87,6 +89,46 @@ def process(self, input):
             Updated for JWST version 1.3.3, code restructure
         '''
         with datamodels.RampModel(input) as input_model:
+
+            if self.s1_meta.update_sat_flags:
+                input_model = update_saturation.update_sat(input_model, 
+                                                           self.s1_log, 
+                                                           self.s1_meta)
+
+            if self.s1_meta.grouplevel_bg:
+                from ..S3_data_reduction import background as bkg
+                import astraeus.xarrayIO as xrio
+                
+                all_data = input_model.data
+                dq = input_model.groupdq
+
+                self.s1_meta.inst = input_model.meta.instrument.name.lower()
+                self.s1_meta.int_start = 0
+                self.s1_meta.n_int = all_data.shape[0]
+
+                for ngrp in range(all_data.shape[1]):
+                    grp_data = all_data[:, ngrp, :, :]
+                    grp_mask = np.ones(grp_data.shape, dtype=bool)
+                    dqmask = np.where((dq[:, ngrp, :, :] % 2 == 1) |
+                                      (dq[:, ngrp, :, :] == 2))
+                    grp_mask[dqmask] = 0
+
+                    xrdata = (['time', 'y', 'x'], grp_data)
+                    xrmask = (['time', 'y', 'x'], grp_mask)                
+                    xrdict = dict(flux=xrdata, mask=xrmask)
+                    data = xrio.makeDataset(dictionary=xrdict)
+                    data['flux'].attrs['flux_units'] = 'n/a'
+                    if ngrp == all_data.shape[1]-1:
+                        data = bkg.BGsubtraction(data, self.s1_meta, 
+                                                 self.s1_log, 
+                                                 self.s1_meta.isplots)
```

Comment:
The 4th input parameter is `m` (The current file/batch number).  `isplots` isn't currently an input parameter; however, I think we should make it one.  We would then need to change all references in BGsubtraction() from `meta.isplots_S3` to `isplots` and modify calls in s3_reduce.py to include the additional parameter.  I think this is an elegant solution and has the added benefit of not needing to define `meta.isplots_S3` in Stage 1.